### PR TITLE
Add typed errors do config module

### PR DESCRIPTION
**Pull request by endersonmaia**
This shall make it easier to implement new tests to the config module.

### CODE REVIEW

File: apps/cli/src/config.ts [hunk 1]
```diff
@@ -3,6 +3,72 @@ import os from "os";
 import { extname } from "path";
 import { TomlPrimitive, parse as parseToml } from "smol-toml";
 
+/**
+ * Typed Errors
+ */
+export class InvalidBuilderError extends Error {
+    constructor(builder: any) {
```
Comment (tuler) [inline apps/cli/src/config.ts:10]: don't use any, use `string`

File: apps/cli/src/config.ts [hunk 2]
```diff
@@ -3,6 +3,72 @@ import os from "os";
...
+        super(`Invalid builder: ${builder}`);
+        this.name = "InvalidBuilder";
+    }
+}
+
+export class InvalidDriveFormatError extends Error {
+    constructor(format: any) {
```
Comment (tuler) [inline apps/cli/src/config.ts:17]: don't use any, use `string`

File: apps/cli/src/config.ts [hunk 3]
```diff
@@ -3,6 +3,72 @@ import os from "os";
...
+        super(`Invalid drive format: ${format}`);
+        this.name = "InvalidDriveFormatError";
+    }
+}
+
+export class InvalidEmptyDriveFormatError extends Error {
+    constructor(format: any) {
```
Comment (tuler) [inline apps/cli/src/config.ts:24]: don't use any, use `string`

**Review by tuler — COMMENTED** [review]
review the usage of `any` in general

File: apps/cli/src/config.ts (hunk 1 above)
Comment (endersonmaia) [reply apps/cli/src/config.ts:10]: When using `string`, the IDE suggest to call it likle `new InvalidBuilderError(true as unknown as string)`, which is weird for me.

What's the best approach in those cases?

**Review by tuler — APPROVED** [review]

**Pull request merged**